RATEST-200: E2E Location workflow test

kdaud opened the pull request:
Ticket ID

Description -> E2E Location workflow test

kdaud commented:
Waiting for review cc: @sherrif10

kdaud commented:
Waiting for the review cc: @sherrif10

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/LocationsSteps.java:
```diff
@@ -0,0 +1,176 @@
+public class LocationsSteps extends Steps {
+
+	private static final String PREFERRED_HANDLER = "Location Field Gen Datatype Handler";
+	private static final String DESCRIPTION = "The location is for expectant mothers";
```
What is the description all about

kdaud replied:
Have rather refactored it into the exact method that uses it.

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/LocationsSteps.java:
```diff
@@ -0,0 +1,176 @@
+	@And("a user enters location tag details into the form")
+	public void enterLocationTagDetails() {
+		manageLocationTagsPage.enterLocationTagName("Maternity Ward Wing");
+		manageLocationTagsPage.enterLocationTagDescription("For expectant mothers");
```
I thought here you would use DESCRIPTION variable

kdaud replied:
Location Tags has its own description that is required when entering its details.

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/Steps.java:
```diff
@@ -28,6 +28,7 @@
 	protected By patientHeaderId = By.cssSelector("div.identifiers span");
 	protected InitialSetupPage initialSetupPage;
 	protected PatientVisitsDashboardPage visitsDashboardPage;
+	protected HomePage homePage;
```
Why including this page here

kdaud replied:
Putting in mind that HomePage will be used by other workflows tests other than Location test, I think we can have it in the base class so that any test that will need it will access it through inheritance logic than declaring it in every test that will use it.

sherrif10 replied:
Are you  calling it anywhere with in steps class?

kdaud replied:
> Are you calling it anywhere with in steps class?

No

sherrif10 replied:
You only need to include what is required i think

kdaud replied:
@sherrif10 did you see the reason [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/131#discussion_r712758416) for declaring the page from the base class?

sherrif10 replied:
its not worth maintaining it there, suppose its does not get used like that at all, Personally, i don't think we need to have that page initialised in steps class

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/LocationsSteps.java:
```diff
@@ -0,0 +1,176 @@
+	}
+
+	@And("a user clicks on save location attribute type button")
+	public void clickOnSaveLocationAttributeTypeButton() {
```
Method naming would sound more cleaner as  clickOnSaveButton since by default it saves location in the behind scenes

kdaud commented:
Waiting for more recommendation to make the workflow test better! cc: @sherrif10

kdaud commented:
Waiting for more recommendations cc: @sherrif10

kdaud commented:
Tagging @dkayiwa for final review is a good thing when @sherrif10 has approved the PR! Not sure whether @sherrif10 has more recommendations otherwise I have addressed the reviews!

kdaud commented:
Waiting for final reviews cc: @dkayiwa

dkayiwa commented on src/test/java/org/openmrs/contrib/qaframework/automation/LocationsSteps.java:
```diff
@@ -0,0 +1,177 @@
+
+	@Then("the system loads manage locations page")
+	public void systemLoadsManageLocationsPage() {
+		assertTrue(textExists("Manage Locations"));
```
Will this work when the locale is switched from English?

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/LocationsSteps.java:
```diff
@@ -0,0 +1,177 @@
+	public void enterLocationAttributeTypeDetails() {
+		attributePage.enterLocationAttributeTypeName("Female Ward");
+		attributePage
+				.enterLocationAttributeTypeDescription("The location is for expectant mothers");
```
Correct this, Probably on the same line  looks better

sherrif10 commented:
Hello @kdaud Any further progress on this ticket, Is there any where we can provide support

kdaud commented:
>Thanks @sherrif10 for the check-in !

I was busy taken up with some other work elsewhere But will clear this workflow over the weekend to have it run in master branch.

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/LocationsSteps.java:
```diff
@@ -0,0 +1,182 @@
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.ConfigureMetadataPage;
```
Change the import statemement package to import from https://github.com/openmrs/openmrs-contrib-qaframework/tree/master/src/test/java/org/openmrs/contrib/qaframework/page to cater for the current changes

sherrif10 commented:
Hello @kdaud are you still working on this ?
